services/src/main/java/org/keycloak/services/resources/admin/RoleMapperResource.java

```diff
@@ -200,7 +200,7 @@ public MappingsRepresentation getRoleMappings() {
 
         return realm.getRolesStream()
                 .filter(this::canMapRole)
-                .filter(((Predicate<RoleModel>) roleMapper::hasRole).negate())
+                .filter(((Predicate<RoleModel>) roleMapper::hasDirectRole).negate())
```

Comment:
See the comment I did for client roles and if makes sense to change hint for available realm roles.